lib/mountain_view.rb

```diff
@@ -1,7 +1,10 @@
-require "mountain_view/version"
-require "mountain_view/configuration"
-require "mountain_view/presenter"
-require "mountain_view/component"
+# frozen_string_literal: true
+
+require 'mountain_view/version'
```

Comment:
More often I've seen a message like the one below "Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols."

My understanding was always use single quotes, unless there was a single quote in the string, or the string had interpolation.  

But the above seems to be the opposite.  Is there more to this code standard than I originally understood?  Is there a resource you can point me to?

Reply:
That's a great question!
There are two popular styles in the Ruby community, both of which are considered good.
The default in Rubocop is to use single quotes unless you need double quotes for interpolation or special characters like `\t`, `\n`.
I prefer to use double quotes by default because it makes the code more consistent, don't have to think if the string has special characters or interpolation, can save a few errors if you forget about special characters, and looks nicer in diffs if you change a plain string to one with interpolation as you don't need to change the quotes but just the string content.

A nice resource about all things rubocop related is the [ruby-style-guide](https://github.com/rubocop-hq/ruby-style-guide#consistent-string-literals) repo— linked straight to the string section. Oh, and of course [rubocop's docs](https://rubocop.readthedocs.io/en/latest/cops_style/#stylestringliterals).

Reply:
Excellent Info, thank  you :)  I like double quotes by default as well now that I've seen that it is an option.  I'll read through the style guide and docs...  This is awesome :)